### PR TITLE
Model of sharpe ratio for single token model

### PR DESCRIPTION
**Pull request by leafyoung**
*Test*

```
credmark-dev run finance.sharpe-ratio-token --input '{"token": {"address": "0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9"}, "return_rolling_days": 180, "risk_free_rate": 0.02}' -b 14234904 --format_json
```

Output

```
"output": {
        "sharpe_ratio": 2.0218432585425856,
        "window": "360 days",
        "blockTime": "2022-02-17 23:59:34",
        "block_number": 14226744,
        "blockTimestamp": 1645142374,
        "token_address": "0x7fc66500c84a76ad7e9c93437bfc5ac33e2ddae9"
    },
```

### CODE REVIEW

File: models/credmark/algorithms/portfolio/sharpe.py [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+from credmark.dto import DTO, DTOField
+from credmark.cmf.types import Token, Price
+
+import numpy as np
+import pandas as pd
+
+
+class SharpRatioInput(DTO):
+    token: Token
+    return_rolling_days: int = DTOField(182, ge=2)
```
Comment (newtectonics) [inline models/credmark/algorithms/portfolio/sharpe.py:12]: interesting to hardcode in the "days" unit. I guess I'm ok with it, but I don't LOVE it ;) could we do a generic dto for interval-based agg stuff?

**Review by newtectonics — APPROVED** [review]

File: models/credmark/algorithms/portfolio/sharpe.py (hunk 1 above)
Comment (leafyoung) [reply models/credmark/algorithms/portfolio/sharpe.py:12]: Ok. got you. I have updated the model input to use generic window. The rolling return can be derived as half the window.

```
credmark-dev run finance.sharpe-ratio-token --input '{"token": {"address": "0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9"}, "window": "10 minute", "risk_free_rate": 0.02}' -b 14234904 --format_json
```

Or,
```
credmark-dev run finance.sharpe-ratio-token --input '{"token": {"address": "0x7Fc66500c84A76Ad7e9c93437bFc5Ac33E2DDaE9"}, "window": "360 days", "risk_free_rate": 0.02}' -b 14234904 --format_json
```

**Review by newtectonics — DISMISSED** [review]
This is inverted. The input should be

token: Token
prices: BlockSeries[Price]
risk_free_rate: float

I'll approve for now, but let's invert this.

**Review by newtectonics — APPROVED** [review]
looks good!

**Pull request merged**